### PR TITLE
NEW: Accountancy - Add mode intra & export for product or service bought

### PR DESCRIPTION
**Pull request by aspangaro**
Need to prepare "auto liquidation"

### CODE REVIEW

File: htdocs/accountancy/admin/productaccount.php [hunk 1]
```diff
@@ -482,11 +528,40 @@
...
+			elseif ($accounting_product_mode == 'ACCOUNTANCY_BUY_INTRA') {
+				$compta_prodbuy = (! empty($conf->global->ACCOUNTING_PRODUCT_BUY_INTRA_ACCOUNT) ? $conf->global->ACCOUNTING_PRODUCT_BUY_INTRA_ACCOUNT : $langs->trans("CodeNotDef"));
+				$compta_prodbuy_id = $aarowid_prodbuy_intra;
+			}
+			elseif ($accounting_product_mode == 'ACCOUNTANCY_BUY_EXPORT') {
+				$compta_prodbuy = (! empty($conf->global->ACCOUNTING_PRODUCT_BUY_EXPORT_ACCOUNT) ? $conf->global->ACCOUNTING_PRODUCT_BUY_EXPORT_ACCOUNT : $langs->trans("CodeNotDef"));
+				$compta_prodbuy_id = $aarowid_prodbuy_export;
+			}
+			else {
+				$compta_prodbuy = (! empty($conf->global->ACCOUNTING_SERVICE_BUY_ACCOUNT) ? $conf->global->ACCOUNTING_SERVICE_BUY_ACCOUNT : $langs->trans("CodeNotDef"));
```
Comment (eldy) [inline htdocs/accountancy/admin/productaccount.php]: product and services are mixed here.
We should have ACCOUNTING_PRODUCT_BUY_ACCOUNT here ?

File: htdocs/accountancy/admin/productaccount.php [hunk 2]
```diff
@@ -482,11 +528,40 @@
...
+				$compta_prodbuy_id = $aarowid_servbuy;
+			}
+		}
+		else {
+			if ($accounting_product_mode == 'ACCOUNTANCY_BUY') {
+				$compta_prodbuy = (! empty($conf->global->ACCOUNTING_SERVICE_BUY_ACCOUNT) ? $conf->global->ACCOUNTING_SERVICE_BUY_ACCOUNT : $langs->trans("CodeNotDef"));
+				$compta_prodbuy_id = $aarowid_prodbuy;
```
Comment (eldy) [inline htdocs/accountancy/admin/productaccount.php]: We are in the section about services  (else of if ($obj->product_type == 0)) so we should use aarowid_servbuy, shouldn't it ?

File: htdocs/accountancy/supplier/list.php [hunk 3]
```diff
@@ -426,28 +428,67 @@
...
+		if (! empty($objp->code_buy_p)) {
+			// Value was defined previously
 		} else {
 			$code_buy_p_notset = 'color:orange';
 		}
 		if (empty($objp->code_buy_l) && empty($objp->code_buy_p)) $code_buy_p_notset = 'color:red';
 
-		// $objp->code_buy_p is now code of product/service
-		// $objp->code_buy_l is now default code of product/service
+		// $objp->code_sell_l is now default code of product/service
```
Comment (eldy) [inline htdocs/accountancy/supplier/list.php]: code_sell_l or code_buy_l ? Can you confirm ?

File: htdocs/accountancy/admin/productaccount.php (hunk 1 above)
Comment (aspangaro) [reply htdocs/accountancy/admin/productaccount.php]: Yes, like the system for sales

File: htdocs/accountancy/admin/productaccount.php (hunk 2 above)
Comment (aspangaro) [reply htdocs/accountancy/admin/productaccount.php]: Error fix on product sell too...

**Pull request merged**